Core/include/Acts/Detector/ComponentBuilderProxy.hpp

```diff
@@ -0,0 +1,168 @@
+
+  /// The boundary type
+  VolumeBounds::BoundsType boundsType = VolumeBounds::eOther;
+
+  /// The boundary values - to be interpreted together with boundary type
+  std::vector<ActsScalar> boundaryValues = {};
+
+  /// The transform
+  Transform3 transform = Transform3::Identity();
+
+  /// Request the builder of this object
+  std::shared_ptr<const IDetectorComponentBuilder> builder() const {
+    if (std::holds_alternative<VolumeProxy>(this->holder)) {
+      return std::get<VolumeProxy>(this->holder).builder;
+    } else {
+      std::runtime_error(
+          "ComponentBuilderProxy::builder() - "
+          "Cannot return a builder for a container proxy yet, "
+          "the detecor setup is not complete.");
+    }
+    return nullptr;
+  }
+
+  /// @brief Create the Root volume proxy
+  ///
+  /// @param n The name of the Root proxy
+  /// @param t The transform of the Root proxy
+  /// @param bt The boundary type of the Root proxy
+  /// @param bv The boundary values of the Root proxy
+  /// @param b The builder of the Root proxy - in this case
+  /// it is a single volume builder case
+  ///
+  /// @return The created Root proxy as a shared_ptr
+  static std::shared_ptr<ComponentBuilderProxy> createRootProxy(
+      std::string n, Transform3 t = Transform3::Identity(),
+      VolumeBounds::BoundsType bt = VolumeBounds::eOther,
+      const std::vector<ActsScalar>& bv = {},
+      std::vector<BinningValue> binning = {},
+      std::shared_ptr<const IDetectorComponentBuilder> b = nullptr) {
+    auto proxy = std::make_shared<ComponentBuilderProxy>(n);
+    proxy->transform = t;
+    proxy->boundsType = bt;
+    proxy->boundaryValues = bv;
+    if (b != nullptr) {
+      proxy->holder = VolumeProxy{b};
+    } else {
+      proxy->holder = ContainerProxy{binning};
+    }
+    return proxy;
+  }
+
+  /// @brief Create a child volume proxy
+  ///
+  /// @param n The name of the child volume proxy
+  /// @param t The transform of the child volume proxy
+  /// @param bt The boundary type of the child volume proxy
+  /// @param bv The boundary values of the child volume proxy
+  /// @param b The builder of the child volume proxy
+  std::shared_ptr<ComponentBuilderProxy> addChildVolumeProxy(
+      std::string n, Transform3 t = Transform3::Identity(),
+      VolumeBounds::BoundsType bt = VolumeBounds::eOther,
+      const std::vector<ActsScalar>& bv = {},
+      std::shared_ptr<const IDetectorComponentBuilder> b = nullptr) {
+    auto proxy = std::make_shared<ComponentBuilderProxy>(n);
+    proxy->transform = t;
+    proxy->boundsType = bt;
+    proxy->boundaryValues = bv;
+    proxy->holder = VolumeProxy{b};
+    if (std::holds_alternative<ContainerProxy>(this->holder)) {
+      std::get<ContainerProxy>(this->holder).children.push_back(proxy);
+    } else {
+      throw std::runtime_error(
+          "ComponentBuilderProxy::addChildVolumeProxy() - "
+          "Cannot add a child volume proxy to a non-container proxy");
+    }
```

Comment:
And here you could do `addChild` which throws for the derived class which cannot have child volume proxies.